440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java

```
@@ -252,10 +258,21 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
               + "CLOUD_PROVIDER, SECRET_MANAGER, CLOUD_COST, ARTIFACTORY, CODE_REPO,  "
               + "MONITORING and TICKETING") @QueryParam(SOURCE_CATEGORY_KEY) ConnectorCategory sourceCategory,
       @QueryParam("version") String version, @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo) {
-    accessControlClient.checkForAccessOrThrow(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
-        Resource.of(ResourceTypes.CONNECTOR, null), VIEW_CONNECTOR_PERMISSION);
+    if (accessControlClient.hasAccess(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
+            Resource.of(ResourceTypes.CONNECTOR, null), VIEW_CONNECTOR_PERMISSION)) {
+      return ResponseDTO.newResponse(getNGPageResponse(connectorService.list(page, size, accountIdentifier,
+          orgIdentifier, projectIdentifier, searchTerm, type, category, sourceCategory, version, emptyList())));
+    }
+    Page<Connector> allConnectors = connectorService.listAll(page, size, accountIdentifier, orgIdentifier,
+        projectIdentifier, searchTerm, type, category, sourceCategory, version);
+
+    List<Connector> permittedConnectors = connectorRbacHelper.getPermitted(allConnectors.getContent());
+    if (isEmpty(permittedConnectors)) {
+      return ResponseDTO.newResponse(getNGPageResponse(Page.empty()));
```

We should throw an error here for user not having access to view connector.